### ISSUE TITLE
Link component can provide it's own  implementation of anchor tag (as per NavigationItem)

### ISSUE THREAD

**Issue by mark-tate** [jpmorganchase/salt-ds#3627]
### Area

UI Components

### The problem

Add render prop/patten to Link component

With NavigationItem, we can now provide an alternate link to the NavigationItem through the `render` prop. Emulate this behaviour for `Link`

### The solution

Review NavigationItem API and render prop and emulate equivalent solution for Link

### Alternatives and examples

- Provider a 'Render Prop' and 'Render Element' example for your implementation
- Update docs inline with how we documented NavigationItem

### Are you a JPMorgan Chase & Co. employee?

- [ ] I am an employee of JPMorgan Chase & Co.

**Comment by saurabhkacholiya** [comment]
Hi @mark-tate I would like to take up this issue. can you please assign it to me?

**Comment by ArchitGajjar** [comment]
Hi @mark-tate - can I work on this issue ?
- Yes, I am an employee of JPMorgan Chase & Co.

**Comment by origami-z** [comment]
> Hi @mark-tate - can I work on this issue ?
> 
> * Yes, I am an employee of JPMorgan Chase & Co.

Yes, feel free to contribute (assuming you've done the internal open source training)

**Comment by ArchitGajjar** [comment]
Sure, thank you @origami-z

**Comment by nidhi752** [comment]
hi, i would love to help in this issue.

**Comment by origami-z** [comment]
> hi, i would love to help in this issue.

Feel free to contribute a PR. If you're not part of JPMC, remember to follow CLA process from  https://github.com/jpmorganchase/.github/blob/main/CONTRIBUTING.md

**Comment by mark-tate** [comment]
Many offers of contribution so hoping that someone will provide the specified solution.
Have a look at NavigationItem's render api, would be a copy of that API.
Use this PR to discuss.

Feel free to contribute a PR. If you're not part of JPMC, remember to follow CLA process from https://github.com/jpmorganchase/.github/blob/main/CONTRIBUTING.md

Our own capacity will pick this up in Q1 otherwise.
